tests/e2e/util/common/e2e_utils.go

```diff
@@ -261,3 +263,46 @@ func InstallOperatorViaHelm(extraArgs ...string) error {
 func UninstallOperator() error {
 	return helm.Uninstall("sail-operator", "--namespace", OperatorNamespace)
 }
+
+// This func will be used to get URLs for the yaml files of the testing apps. Example: helloworld, sleep, tcp-echo.
+// Default values points to upstream Istio sample yaml files.
+// The base URL can be overridden by setting the environment variable POD_SAMPLE_YAML_BASE_URL.
+
+// GetYAMLPodURL returns the URL of the yaml file for the testing app.
+// args:
+// version: the version of the Istio to get the yaml file from.
+// appName: the name of the testing app. Example: helloworld, sleep, tcp-echo.
+func GetYAMLPodURL(version supportedversion.VersionInfo, appName string) string {
+	var url string
+
+	switch appName {
+	case "tcp-echo-dual-stack":
+		url = "samples/tcp-echo/tcp-echo-dual-stack.yaml"
+	case "tcp-echo-ipv4":
+	case "tcp-echo":
+		url = "samples/tcp-echo/tcp-echo-ipv4.yaml"
+	case "tcp-echo-ipv6":
+		url = "samples/tcp-echo/tcp-echo-ipv6.yaml"
+	case "sleep":
+		url = "samples/sleep/sleep.yaml"
+	case "helloworld":
+	case "sample":
+		url = "samples/helloworld/helloworld.yaml"
+	default:
+		return ""
+	}
+
+	// Get the base URL from the environment variable or use the default
+	baseURL := os.Getenv("POD_SAMPLE_YAML_BASE_URL")
```

Comment:
Does it mean, that if I'm setting a custom "baseURL" based on the env var, the path for the yaml files should be identical to the mentioned in lines 279-290?

Reply:
Yes, I'll improve this to be able to set the full path

Reply:
@MaxBab take a look, I think the best way is to set each yaml individually. With this, wdyt?

Reply:
@fjglira 

Cool!
I like this approach.